### PR TITLE
separate the WS server from introspector; multiple listen addrs.

### LINKED ISSUE #6: Have a strategy to select an appropriate Listen Address for the server
The caller passes multiple ListenAddresses when starting the Introspection server. We need a way to be able to determine an  appropriate one to use among them rather than just using the first address in the list like we do now. Please see https://github.com/libp2p/go-libp2p-introspection/pull/1#discussion_r376008248

### PR DESCRIPTION
**Pull request by raulk**
Fixes #6.

* This PR separates the introspector itself from the endpoint/server.
* It adds Start() and Close() methods to control the lifecycle of the server.
* `ListenAddrs()` now returns the actual listen socket address, not the user-provided address, e.g. we return `127.0.0.1:<actual dynamic port>` instead of `127.0.0.1:0` if the user passed 0 as a port. Not sure what happens for BIND_ALL addresses, though.
* We now support multiple listen addresses.
* Added unit tests for the server.

### CODE REVIEW

**Comment by raulk** [comment]
@aarshkshah1992 merged this because I need to make a release to integrate downstream (go-libp2p).

File: introspector.go [hunk 1]
```diff
@@ -93,7 +89,7 @@ func (d *DefaultIntrospector) FetchFullState() (*introspectpb.State, error) {
 	if d.tree.Traffic != nil {
 		tr, err := d.tree.Traffic()
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to fetch traffic")
+			return nil, fmt.Errorf("failed to fetch traffic: %w", err)
 		}
```
Comment (aarshkshah1992) [inline introspector.go:93]: @raulk Out of curiosity, why is this preferred ?

File: ws_server.go [hunk 2]
```diff
@@ -1,48 +1,117 @@
...
+
+	if len(s.listeners) > 0 {
+		return errors.New("failed to start WS server: already started")
+	}
 
-	// start server
-	srv := http.Server{
-		// TODO Need a better strategy to select an address
-		Addr: introspector.ListenAddrs()[0],
+	if len(s.config.ListenAddrs) == 0 {
```
Comment (aarshkshah1992) [inline ws_server.go:57]: @raulk A nil config will panic here as we don't check for this in `NewWsServer `.

File: ws_server.go [hunk 3]
```diff
@@ -1,48 +1,117 @@
...
-	}()
 
-	logger.Infof("server starting, listening on %s", introspector.ListenAddrs()[0])
+		s.closeWg.Add(1)
+		go func() {
+			if err := s.server.Serve(l); err != http.ErrServerClosed {
+				logger.Errorf("failed to start WS server, err: %s", err)
+			}
+			s.closeWg.Done()
+		}()
```
Comment (aarshkshah1992) [inline ws_server.go:75]: @raulk We should also wait on the WaitGroup when we close the server.

**Pull request merged**